src/xtopcom/xpreprocess/src/xpreprocessor.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2023-present Telos Foundation & contributors
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "xplugin/xplugin_manager.h"
+#include "xpreprocess/xpreprocess.h"
+
+#include <iostream>
+using namespace std;
+
+NS_BEG2(top, data)
+
+xpreprocess & xpreprocess::instance() {
+    static xpreprocess ins;
+    return ins;
+}
+
+xpreprocess::xpreprocess() {
+    m_handler = std::make_shared<xplugin_manager>();
+}
+
+bool xpreprocess::send(xmessage_t & msg) {
+    if (m_handler == nullptr) {
```

Comment:
since m_handle is initialized in constructor, why add `if (m_handler == nullptr)` check? it must be non-null.